src/test/kotlin/no/brreg/informasjonsforvaltning/abackendservice/utils/TestUtils.kt

```diff
@@ -57,22 +27,23 @@ fun getContent(host: String, port: Int, address: String): Map<String,Any> {
 }
 
 
-fun simplePost(host: String, port: Int, address: String,body: String? = null, token: String? = null): Map<String, String> {
-    val connection = URL("http", host, port, address).openConnection() as HttpURLConnection
+fun apiPost(endpoint : String, body: String?, token: String?): Map<String, String> {
+    val connection  = URL(getApiAddress(endpoint)).openConnection() as HttpURLConnection
     connection.requestMethod = "POST"
     connection.setRequestProperty("Content-type", "application/json")
     connection.setRequestProperty("Accept", "application/json")
-    if(token!= null) connection.setRequestProperty("Authorization", "Bearer {$token}")
+
+    if(!token.isNullOrEmpty()) {connection.setRequestProperty("Authorization", "Bearer $token")}
 
     try {
+        if (body!=null) {
+            connection.doOutput = true
+            val writer = OutputStreamWriter(connection.outputStream)
+            writer.write(body)
+            writer.close()
 
-    if (body!=null) {
-        connection.doOutput = true
+        }
 
-        connection.outputStream.bufferedWriter().write(body)
-        connection.outputStream.flush()
-        connection.outputStream.close()
-    }
         val response = mapOf<String,String>(
```

Comment:
Ved å bruke "connection.connect()" kan du sjekke responseCode før du prøver å lese body osv, tror det var bakgrunnen for feilen jeg fikk på get-metoden i går